ONSdigital/dp-frontend-search-controller
# Filter previous releases

### What

The task is here: https://jira.ons.gov.uk/browse/DIS-1293 and is described as follows:

Now that the Search API has functionality to allow filtering by uri_prefix, when rendering /previousReleases we should pass this as a parameter in the search query. This is a modification to the dp-frontend-search-controller.

Request for:

`/economy/bulletins/myGreatBulletin/previousReleases`

Should pass `/economy/bulletins/myGreatBulletin/` to the Search API to filter the results.  

### How to review

To test it locally, but pointing to the API Router in sandbox, do as follows:

- Log into aws sandbox and do a dp remote allow:
`aws sso login --profile=dp-sandbox`
`dp remote allow sandbox`
- Run the dp-frontend-search-controller:
`make debug API_ROUTER_URL=https://api.dp.aws.onsdigital.uk/v1`
- Run the dp-frontend-router:
`make debug API_ROUTER_URL=https://api.dp.aws.onsdigital.uk/v1 PREVIOUS_RELEASES_ROUTE_ENABLED=true`
- Run the dp-design-system:
`make debug`
- You should then see the list of filtered previous releases at any suitable link e.g.

http://localhost:20000/businessindustryandtrade/changestobusiness/mergersandacquisitions/bulletins/mergersandacquisitionsinvolvingukcompanies/previousreleases

Note that if you currently go to the same URL path in sandbox then the list of previous releases will be unfiltered (so many more results are shown) e.g. 

https://dp.aws.onsdigital.uk/businessindustryandtrade/changestobusiness/mergersandacquisitions/bulletins/mergersandacquisitionsinvolvingukcompanies/previousreleases

### Who can review

Anyone.
